nostr-client-plus/src/bin/aggregator.rs

```diff
@@ -296,62 +303,91 @@ async fn run() -> Result<()> {
                 continue;
             }
 
-            match aggr_book.entry(job_id.clone()) {
-                Entry::Occupied(mut entry) => {
-                    tracing::debug!("Another result for job {} found", job_id);
-                    let (workers, payload) = entry.get_mut();
+            // Falling below are logics for classification
+            let raw_data_id = hex::encode(new_payload.header.raw_data_id.hash());
+            let mut fetch_connection = redis_client.get_connection();
```

Comment:
I am not sure it's efficient to create a redis connection in a loop, one for every operation we are supposed to do.
Why cannot you re-use the same connection as we did before. Redis was already used in a loop, not sure what's changed this time. But a connection for every message we receive is not the way forward.

You tried to re-use `redis_con` and because it's used in another task you got an error. The solution is not to create 3 for every loop (I see that you do the same for delete and update..), but to have 2 separate connections, one that goes in one task (the original one where it already was), and another one in your task.

So, go above, create an addition `redis_con_agg`  and use that (with `&mut`  or `&`) in every place where you have been creating a new redis connection inside the loop

Reply:
use a new redis_con_task connection now